apps/admin-gui/src/app/shared/components/dialogs/add-role-dialog/add-role-dialog.component.ts

```diff
@@ -54,9 +58,15 @@ export class AddRoleDialogComponent implements OnInit {
     this.selectedRule = this.rules[0];
     if (this.rules.some((rule) => rule.primaryObject === 'Facility')) {
       // Not callable by SELF, need to check privilege
-      this.facilityService
-        .getAllFacilities()
-        .subscribe({ next: (facilities) => (this.facilities = facilities) });
+      this.facilityService.getAllFacilities().subscribe({
+        next: (facilities) =>
+          (this.facilities = facilities.map((fac) => {
+            const ef: EnrichedFacility = {
+              facility: fac,
+            };
+            return ef;
+          })),
```

Comment:
You are able to reuse existing pipe also in ts file.
```suggestion
          (this.facilities = new ToEnrichedFacilityPipe().transform(facilities)),
```